.github/workflows/action-e2e.yaml

```diff
@@ -0,0 +1,147 @@
+# End-to-end tests for the Publish to BCR custom GitHub action
+name: action-e2e
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+concurrency:
+  # Cancel previous actions from the same PR or branch except 'main' branch.
+  # See https://docs.github.com/en/actions/using-jobs/using-concurrency and https://docs.github.com/en/actions/learn-github-actions/contexts for more info.
+  group: concurrency-group::${{ github.workflow }}::${{ github.event.pull_request.number > 0 && format('pr-{0}', github.event.pull_request.number) || github.ref_name }}${{ github.ref_name == 'main' && format('::{0}', github.run_id) || ''}}
+  cancel-in-progress: ${{ github.ref_name != 'main' }}
+jobs:
+  # Each job is an e2e test. Unfortunately a full workflow cannot be added as
+  # required status check on branch protection rules, so each test must be added
+  # as a required check: https://github.com/orgs/community/discussions/12395.
+  test-happy-path:
+    # Create a new module entry and test the content
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          path: this
+      - name: Setup test fixture
+        run: |
+          set -o errexit -o nounset -o pipefail -o xtrace
+
+          # Create a release archive from a fixture
+          tar -czvf archive.tar.gz -C this/e2e/fixtures/versioned --transform 's,^./,versioned-1.0.0/,' --sort=name --owner=root:0 --group=root:0 --mtime='UTC 1980-02-01' .
+
+          # Substitute the archive url to a local file path
+          cat this/e2e/fixtures/versioned/.bcr/source.template.json | jq ".url = \"file://$(realpath archive.tar.gz)\"" > "/tmp/source.template.json"
+          mv /tmp/source.template.json this/e2e/fixtures/versioned/.bcr/source.template.json
+          cat this/e2e/fixtures/versioned/.bcr/source.template.json
+      - name: Create registry
+        run: |
+          mkdir -p bazel-central-registry/modules
+          cd bazel-central-registry
+          git init
+      - name: Create entry
+        uses: ./this
+        with:
+          tag: v1.0.0
+          module-version: 1.0.0
+          github-repository: foobar/versioned
+          templates-dir: this/e2e/fixtures/versioned/.bcr
+          local-registry: bazel-central-registry
+      - name: Test entry content
+        run: |-
+          set -o errexit -o nounset -o pipefail -o xtrace
```

Comment:
maybe it would be easier to read as a standalone shell script instead of embedding in yaml?